### PR TITLE
Better images for illustrating neural networks

### LINKED ISSUE #111: Introduce bias terms
In episode 4, we take a close look at the number of parameters of the different type of layers. To understand where the number of parameters comes from, it is important to know that neurons have a bias term, in addition to the weight parameters. The bias term is not mentioned in episode 4. I think it should either be introduced there, or episode 4 needs a recap of the calculation within a neuron, also explaining the bias term.

### PR DESCRIPTION
**Pull request by colinsauze**
Added a new image to show a single neuron and explain bias terms, fixes #111

Replaced images of the neural network (from #121), deep network and AI/ML/DL differences. 
Do we want to introduce any other images from the slide deck? Perhaps the segmentation image or gradient descent ones?

Do you think the AI/ML/DL image is best in the first section on Deep Learning, Machine Learning and Artificial Intelligence, which is before we've really explained what a neural network and deep learning are or is it better in the neural networks section after we've explained what deep learning is?

### CODE REVIEW

File: _episodes/01-introduction.md [hunk 1]
```diff
@@ -32,23 +32,35 @@ keypoints:
...
+A neural network is an artificial intelligence technique loosely based on the way neurons in the brain work. A neural network consists of connected computational units called neurons. Each neuron has one or more inputs and takes the sum of these each multiple by a weight, performs some, typically non-linear, calculation on the sum and produces one output. This calculation is called the activation function. An extra input called a bias term is usually attached with a value fixed at 1.
```
Comment (psteinb) [inline _episodes/01-introduction.md:43]: This could be reframed as a bullet list for easier visual parsing. Perhaps like so:

```suggestion
A neural network is an artificial intelligence technique loosely based on the way neurons in the brain work. A neural network consists of connected computational units called **neurons**. Each neuron ... 

- has one or more inputs, e.g. input data expressed as floating point numbers
- most of the time, each neuron conducts 3 main operations:
  + take the weighted sum of the inputs
  + add an extra constant weight (i.e. a bias term) to this weighted sum
  + apply a non-linear function to the output so far (using a predefined activation function)
- return one output value, again a floating point number
```

This is just a suggestion.

File: _episodes/01-introduction.md [hunk 2]
```diff
@@ -32,23 +32,35 @@ keypoints:
...
+![A diagram of a single artificial neuron](../fig/01_neuron.png){: width="600px"}
 
-![An infographics showing the relation of AI, ML, NN and DL](../fig/AI_ML_DL_bubble_square_draft.png){: width="600px" }
+Multiple neurons can be joined together by connecting the output of one to the input of another. These connections are called edges. The weights determines the 'strength' of the connection, the weights are adjusted during training. In this way, the combination of neurons and edges describe a computational graph, an example can be seen in the image below. In most neural networks neurons are aggregated into layers. Signals travel from the input layer to the output layer, possibly through one or more intermediate layers called hidden layers.
```
Comment (psteinb) [inline _episodes/01-introduction.md:47]: This paragraph mentions the term `edge` multiple times. Will this be reused later? If not, I'd contemplate to not stress it too much to keep the cognitive load low.

File: _episodes/01-introduction.md [hunk 3]
```diff
@@ -32,23 +32,35 @@ keypoints:
...
+
+Deep Learning requires extensive training using example data which shows the network what output it should produce for a given input. One common application of Deep Learning is classifying images. Here the network will be trained by being "shown" a series of images and told what they contain. Once the network is trained it should be able to take another image and correctly classify its contents. But we are not restricted to just using images, any kind of data can be learned by a Deep Learning neural network. This makes them able to appear to learn a set of complex rules only by being shown what the inputs and outputs of those rules are instead of being taught the actual rules. Using these approaches Deep Learning networks have been taught to play video games and even drive cars. The data on which networks are trained usually has to be quite extensive, typically including thousands of examples. For this reason they are not suited to all applications and should be considered just one of many machine learning techniques which are available.
```
Comment (psteinb) [inline _episodes/01-introduction.md:56]:
```suggestion
Deep Learning requires extensive training using large example datasets. During training, the network is shown what output it should produce for a given input. One common application of Deep Learning is classifying images. Here the network will be trained by being "shown" a series of images and told what they contain. Once the network is trained it should be able to take another (unseen) image and correctly classify its contents. But we are not restricted to just using images, any kind of data can be learned by a Deep Learning neural network. This makes them able to appear to learn a set of complex rules only by being shown what the inputs and outputs of those rules are instead of being taught the actual rules. Using these approaches Deep Learning networks have been taught to play video games and even drive cars. The data on which networks are trained usually has to be quite extensive, typically including thousands of examples. For this reason they are not suited to all applications and should be considered just one of many machine learning techniques which are available.
```

I really like this paragraph as in touches on so many things between the lines.

**Review by psteinb — CHANGES_REQUESTED** [review]

**Review by dafnevk — COMMENTED** [review]
Looks good to me!
I like the AI/ML/DL image situated where it is right now, it's very generic and after that we dive into the details.
I think the search landscape image could be used in addition to or instead of the loss function graph that is now under step 6. Or maybe already in step 5.

**Comment by colinsauze** [comment]
I think I have resolved all issues with this pull request and would like to get it merged before Monday's workshop at EMBL. Does anyone have any issues with it being merged?

**Review by psteinb — APPROVED** [review]
LGTM! Thanks a bunch for these wonderful contributions.
